beacon-chain/sync/validators_custody.go

```diff
@@ -35,14 +35,14 @@ func (s *Service) setTargetValidatorsCustodyRequirement() {
 	}
 
 	// Retrieve the head state.
-	headState, err := s.cfg.chain.HeadStateReadOnly(s.ctx)
-	if err != nil || headState == nil {
-		log.WithError(err).Error("Failed to get head state")
+	finalizedState := s.cfg.stateGen.FinalizedState()
+	if finalizedState == nil || finalizedState.IsNil() {
+		log.Error("Finalized state is nil")
 		return
 	}
 
 	// Get the validators custody requirement.
-	validatorsCustodyRequirement, err := peerdas.ValidatorsCustodyRequirement(headState, indices)
+	validatorsCustodyRequirement, err := peerdas.ValidatorsCustodyRequirement(finalizedState, indices)
```

Comment:
can't we change this to a read only state?

Reply:
No need actually.
Fixed in 3c888de93bc9beb83f918532129b604e8ce05146.